### ISSUE TITLE
Installer doesn't boot on Hyper-V

### ISSUE THREAD

**Issue by puneetse** [clearlinux/clr-installer#465]
**Describe the bug**
The live-server.iso does not boot on Hyper-V and gets stuck at "Searching for installer media, retrying...".

**To Reproduce**
1. Create Hyper-V VM
2. Add a DVD Drive to the VM
3. Mount live-server.iso (tested with 30210 and 29690)
4. Try to boot from it.

**Expected behavior**
The installer ISO should boot under Hyper-V VMs

**Additional context**
The bootloader appears so the DVD drive is there. On a working Hyper-V VM with CL, the same device shows up as `/dev/sr0` with the `CLR_ISO` label when the ISO is mounted.

**Comment by mdhorn** [comment]
@puneetse Does Hyper-V only run under windows? If so, what version? Where do we get Hyper-V?
@evil-brain Do you still have a Windows system?

**Comment by puneetse** [comment]
Yes, Hyper-V is a Window thing. I tried server 2016 and server 2019. I'm sure the Win 10 Hyper-V client would do the same thing. 

I can help source a system to debug on if needed.

**Comment by evil-brain** [comment]
I've done some investigation and here's where I currently am:
The installer ISO is built using a  kernel that currently doesn't have support for virtual hardware under Hyper-V. We do have a kernel with support - it's called 'linux-hyperv'.

I built a custom installer using this kernel, but ran into another issue - the ATA_PIIX driver is built as a module and not built into the kernel. Because of this the intird cannot work - as it can't find the CD Drive to load from.

Here's an article from Microsoft about a similar issue: https://support.microsoft.com/en-us/help/2600152/cd-rom-device-is-no-longer-accessible-after-installing-the-hyper-v-lin

Checking the kernel configs, the currently-used kernel has the ata_piix driver built in, but the hyper-v kernel builds this driver as a module. This cannot work for ISO booting.

So I think we need to change the config, rebuild and change the kernel, and build a custom installer image to support Hyper-V.

**Comment by evil-brain** [comment]
Found a weird solution. You'll have to build your own image:
 * Install the clr-installer installer bundle

* Download the yaml: 
https://download.clearlinux.org/current/config/image/live-server.yaml (or live-desktop)
download all the related scripts as well (may be easier to just downlaod all .sh files to the same directory)

* run the command:
`sudo clr-insatller --keep-image --config <path/to/live-server.yaml>`

Now you have live-server.img - but we need to convert this to a vhdx file for Hyper-V to work.
* Use qemu-img (requires qemu bundle):
`qemu-img convert -f raw -O vhdx live-server.img live-server.vhdx`

Copy live-server.vhdx to a USB drive or otherwise move to the Windows machine.
Now, a weird workaround must be done. If this workaround isn't performed, you'll get this error when attempting to boot:
"The version does not support this version of the file format."

* The workaround is to copy/paste the .vhdx file.

Seriously. Here's an article: https://www.itexperience.net/2014/05/28/the-version-does-not-support-this-version-of-the-file-format-in-hyper-v-manager/

I don't know why it works, but it does (comparing hex dumps shows lots of changes)

In Hyper-V manager, point the virtual HDD to this copy of the image file, and boot.

**Comment by tristanbarcelon** [comment]
@evil-brain I followed your instruction and it failed towards the latter part of image creation step with this error:

![image](https://user-images.githubusercontent.com/5810165/62815126-dc6d3d80-bae3-11e9-9575-ebd633679fa7.png)

I am performing this step from inside a Hyper-V VM guest running clear linux 30730. This VM guest was created using the available [hyper-v image](https://cdn.download.clearlinux.org/releases/30730/clear/clear-30730-hyperv.vhdx.gz). Did you perform your steps from a bare-metal install of clear linux?

**Comment by tristanbarcelon** [comment]
@evil-brain , I think I have it figured out now after opening up live-desktop.yaml and copying the scripts referenced. I now need to expand the vhdx because it ran out of space :)

**Comment by mdhorn** [comment]
I'm going to close this due to inactivity.
